sys/posix/include/fcntl.h

```diff
@@ -0,0 +1,75 @@
+/*
+ * Copyright (C) 2016 Eistec AB
+ *
+ * This file is subject to the terms and conditions of the GNU Lesser
+ * General Public License v2.1. See the file LICENSE in the top level
+ * directory for more details.
+ */
+
+/**
+ * @file
+ * @ingroup posix
+ * @brief   POSIX compatible fcntl.h definitions
+ * @author  Joakim Nohlgård <joakim.nohlgard@eistec.se>
+ *
+ * @see http://pubs.opengroup.org/onlinepubs/9699919799/basedefs/fcntl.h.html
+ */
+
+#ifndef DOXYGEN
+/* If building on native we need to use the system libraries instead */
+#ifdef CPU_NATIVE
+#pragma GCC system_header
+/* without the GCC pragma above #include_next will trigger a pedantic error */
+#include_next <fcntl.h>
+#else
+#ifndef FCNTL_H_
+#define FCNTL_H_
+
+#include <sys/types.h> /* for mode_t, off_t */
+
+#ifdef __cplusplus
+extern "C" {
+#endif
+
+/* Constants used for the flags in fcntl and open */
+/* The names of the constants are defined by POSIX but most of these numbers
+ * were chosen somewhat arbitrarily */
```

Comment:
Added an `#if` block to include the newlib-provided fcntl.h if building against newlib. Could you test it and see if that fixes your issues?
Especially the O_CREAT, O_TRUNC, O_EXCL flags were different than what newlib has defined.